tracer/src/Datadog.Trace/ClrProfiler/AutoInstrumentation/IbmMq/IbmMqHeadersAdapter.cs

```diff
@@ -5,21 +5,17 @@
 
 #nullable enable
 
+using System;
 using System.Collections.Generic;
+using System.Text;
 using Datadog.Trace.Headers;
 using Datadog.Trace.Util;
 
 namespace Datadog.Trace.ClrProfiler.AutoInstrumentation.IbmMq;
 
-internal readonly struct IbmMqHeadersAdapter : IHeadersCollection
+internal readonly struct IbmMqHeadersAdapter(IMqMessage message) : IHeadersCollection
```

Comment:
Please don't do this - this implicitly captures the variable which changes allocation and layout characteristics. We're adding an analyzer to block this here: https://github.com/DataDog/dd-trace-dotnet/pull/5276